INFO:werkzeug: * Running on http://127.0.0.1:5000/
ERROR:__main__:Exception on / [GET]
Traceback (most recent call last):
  File "c:\Users\Alex\.virtualenvs\lpthw\lib\site-packages\flask\app.py", line 1817, in wsgi_app
    response = self.full_dispatch_request()
  File "c:\Users\Alex\.virtualenvs\lpthw\lib\site-packages\flask\app.py", line 1477, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "c:\Users\Alex\.virtualenvs\lpthw\lib\site-packages\flask\app.py", line 1381, in handle_user_exception
    reraise(exc_type, exc_value, tb)
  File "c:\Users\Alex\.virtualenvs\lpthw\lib\site-packages\flask\_compat.py", line 33, in reraise
    raise value
  File "c:\Users\Alex\.virtualenvs\lpthw\lib\site-packages\flask\app.py", line 1475, in full_dispatch_request
    rv = self.dispatch_request()
  File "c:\Users\Alex\.virtualenvs\lpthw\lib\site-packages\flask\app.py", line 1461, in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
  File "app.py", line 12, in index
    return render_template('index.html', greeting=greeting)
NameError: global name 'render_template' is not defined
INFO:werkzeug:127.0.0.1 - - [06/Aug/2014 00:26:24] "GET / HTTP/1.1" 500 -
INFO:werkzeug: * Running on http://127.0.0.1:5000/
ERROR:__main__:Exception on / [GET]
Traceback (most recent call last):
  File "c:\Users\Alex\.virtualenvs\lpthw\lib\site-packages\flask\app.py", line 1817, in wsgi_app
    response = self.full_dispatch_request()
  File "c:\Users\Alex\.virtualenvs\lpthw\lib\site-packages\flask\app.py", line 1477, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "c:\Users\Alex\.virtualenvs\lpthw\lib\site-packages\flask\app.py", line 1381, in handle_user_exception
    reraise(exc_type, exc_value, tb)
  File "c:\Users\Alex\.virtualenvs\lpthw\lib\site-packages\flask\_compat.py", line 33, in reraise
    raise value
  File "c:\Users\Alex\.virtualenvs\lpthw\lib\site-packages\flask\app.py", line 1475, in full_dispatch_request
    rv = self.dispatch_request()
  File "c:\Users\Alex\.virtualenvs\lpthw\lib\site-packages\flask\app.py", line 1461, in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
  File "app.py", line 12, in index
    return render_template('index.html', greeting=greeting)
  File "c:\Users\Alex\.virtualenvs\lpthw\lib\site-packages\flask\templating.py", line 127, in render_template
    return _render(ctx.app.jinja_env.get_or_select_template(template_name_or_list),
  File "c:\Users\Alex\.virtualenvs\lpthw\lib\site-packages\jinja2\environment.py", line 830, in get_or_select_template
    return self.get_template(template_name_or_list, parent, globals)
  File "c:\Users\Alex\.virtualenvs\lpthw\lib\site-packages\jinja2\environment.py", line 791, in get_template
    return self._load_template(name, self.make_globals(globals))
  File "c:\Users\Alex\.virtualenvs\lpthw\lib\site-packages\jinja2\environment.py", line 765, in _load_template
    template = self.loader.load(self, name, globals)
  File "c:\Users\Alex\.virtualenvs\lpthw\lib\site-packages\jinja2\loaders.py", line 125, in load
    code = environment.compile(source, name, filename)
  File "c:\Users\Alex\.virtualenvs\lpthw\lib\site-packages\jinja2\environment.py", line 554, in compile
    self.handle_exception(exc_info, source_hint=source)
  File "c:\Users\Alex\.virtualenvs\lpthw\lib\site-packages\jinja2\environment.py", line 742, in handle_exception
    reraise(exc_type, exc_value, tb)
  File "c:\Users\Alex\.virtualenvs\lpthw\lib\site-packages\jinja2\_compat.py", line 36, in reraise
    raise value.with_traceback(tb)
  File "c:\Users\Alex\Dropbox\projects\learn_python_the_hard_way\projects\gothonweb\templates\index.html", line 13, in <module>
    {% endif }
  File "c:\Users\Alex\.virtualenvs\lpthw\lib\site-packages\jinja2\environment.py", line 459, in _parse
    return Parser(self, source, name, encode_filename(filename)).parse()
  File "c:\Users\Alex\.virtualenvs\lpthw\lib\site-packages\jinja2\parser.py", line 893, in parse
    result = nodes.Template(self.subparse(), lineno=1)
  File "c:\Users\Alex\.virtualenvs\lpthw\lib\site-packages\jinja2\parser.py", line 875, in subparse
    rv = self.parse_statement()
  File "c:\Users\Alex\.virtualenvs\lpthw\lib\site-packages\jinja2\parser.py", line 122, in parse_statement
    return getattr(self, 'parse_' + self.stream.current.value)()
  File "c:\Users\Alex\.virtualenvs\lpthw\lib\site-packages\jinja2\parser.py", line 210, in parse_if
    drop_needle=True)
  File "c:\Users\Alex\.virtualenvs\lpthw\lib\site-packages\jinja2\parser.py", line 165, in parse_statements
    next(self.stream)
  File "c:\Users\Alex\.virtualenvs\lpthw\lib\site-packages\jinja2\lexer.py", line 349, in __next__
    self.current = next(self._iter)
  File "c:\Users\Alex\.virtualenvs\lpthw\lib\site-packages\jinja2\lexer.py", line 552, in wrap
    for lineno, token, value in stream:
  File "c:\Users\Alex\.virtualenvs\lpthw\lib\site-packages\jinja2\lexer.py", line 677, in tokeniter
    filename)
jinja2.exceptions.TemplateSyntaxError: unexpected '}'
INFO:werkzeug:127.0.0.1 - - [06/Aug/2014 00:27:12] "GET / HTTP/1.1" 500 -
ERROR:__main__:Exception on / [GET]
Traceback (most recent call last):
  File "c:\Users\Alex\.virtualenvs\lpthw\lib\site-packages\flask\app.py", line 1817, in wsgi_app
    response = self.full_dispatch_request()
  File "c:\Users\Alex\.virtualenvs\lpthw\lib\site-packages\flask\app.py", line 1477, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "c:\Users\Alex\.virtualenvs\lpthw\lib\site-packages\flask\app.py", line 1381, in handle_user_exception
    reraise(exc_type, exc_value, tb)
  File "c:\Users\Alex\.virtualenvs\lpthw\lib\site-packages\flask\_compat.py", line 33, in reraise
    raise value
  File "c:\Users\Alex\.virtualenvs\lpthw\lib\site-packages\flask\app.py", line 1475, in full_dispatch_request
    rv = self.dispatch_request()
  File "c:\Users\Alex\.virtualenvs\lpthw\lib\site-packages\flask\app.py", line 1461, in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
  File "app.py", line 12, in index
    return render_template('index.html', greeting=greeting)
  File "c:\Users\Alex\.virtualenvs\lpthw\lib\site-packages\flask\templating.py", line 127, in render_template
    return _render(ctx.app.jinja_env.get_or_select_template(template_name_or_list),
  File "c:\Users\Alex\.virtualenvs\lpthw\lib\site-packages\jinja2\environment.py", line 830, in get_or_select_template
    return self.get_template(template_name_or_list, parent, globals)
  File "c:\Users\Alex\.virtualenvs\lpthw\lib\site-packages\jinja2\environment.py", line 791, in get_template
    return self._load_template(name, self.make_globals(globals))
  File "c:\Users\Alex\.virtualenvs\lpthw\lib\site-packages\jinja2\environment.py", line 765, in _load_template
    template = self.loader.load(self, name, globals)
  File "c:\Users\Alex\.virtualenvs\lpthw\lib\site-packages\jinja2\loaders.py", line 125, in load
    code = environment.compile(source, name, filename)
  File "c:\Users\Alex\.virtualenvs\lpthw\lib\site-packages\jinja2\environment.py", line 554, in compile
    self.handle_exception(exc_info, source_hint=source)
  File "c:\Users\Alex\.virtualenvs\lpthw\lib\site-packages\jinja2\environment.py", line 742, in handle_exception
    reraise(exc_type, exc_value, tb)
  File "c:\Users\Alex\.virtualenvs\lpthw\lib\site-packages\jinja2\_compat.py", line 36, in reraise
    raise value.with_traceback(tb)
  File "c:\Users\Alex\Dropbox\projects\learn_python_the_hard_way\projects\gothonweb\templates\index.html", line 13, in <module>
    {% endif }
  File "c:\Users\Alex\.virtualenvs\lpthw\lib\site-packages\jinja2\environment.py", line 459, in _parse
    return Parser(self, source, name, encode_filename(filename)).parse()
  File "c:\Users\Alex\.virtualenvs\lpthw\lib\site-packages\jinja2\parser.py", line 893, in parse
    result = nodes.Template(self.subparse(), lineno=1)
  File "c:\Users\Alex\.virtualenvs\lpthw\lib\site-packages\jinja2\parser.py", line 875, in subparse
    rv = self.parse_statement()
  File "c:\Users\Alex\.virtualenvs\lpthw\lib\site-packages\jinja2\parser.py", line 122, in parse_statement
    return getattr(self, 'parse_' + self.stream.current.value)()
  File "c:\Users\Alex\.virtualenvs\lpthw\lib\site-packages\jinja2\parser.py", line 210, in parse_if
    drop_needle=True)
  File "c:\Users\Alex\.virtualenvs\lpthw\lib\site-packages\jinja2\parser.py", line 165, in parse_statements
    next(self.stream)
  File "c:\Users\Alex\.virtualenvs\lpthw\lib\site-packages\jinja2\lexer.py", line 349, in __next__
    self.current = next(self._iter)
  File "c:\Users\Alex\.virtualenvs\lpthw\lib\site-packages\jinja2\lexer.py", line 552, in wrap
    for lineno, token, value in stream:
  File "c:\Users\Alex\.virtualenvs\lpthw\lib\site-packages\jinja2\lexer.py", line 677, in tokeniter
    filename)
jinja2.exceptions.TemplateSyntaxError: unexpected '}'
INFO:werkzeug:127.0.0.1 - - [06/Aug/2014 00:28:11] "GET / HTTP/1.1" 500 -
INFO:werkzeug:127.0.0.1 - - [06/Aug/2014 00:29:11] "GET / HTTP/1.1" 200 -
INFO:werkzeug: * Running on http://127.0.0.1:5000/
INFO:werkzeug:127.0.0.1 - - [06/Aug/2014 16:25:42] "GET / HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [06/Aug/2014 17:12:23] "GET / HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [06/Aug/2014 17:12:33] "GET / HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [06/Aug/2014 17:12:50] "GET / HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [06/Aug/2014 17:13:00] "GET / HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [06/Aug/2014 17:24:47] "GET /hello.html HTTP/1.1" 404 -
INFO:werkzeug: * Running on http://127.0.0.1:5000/
INFO:werkzeug:127.0.0.1 - - [06/Aug/2014 17:25:09] "GET /hello.html HTTP/1.1" 404 -
INFO:werkzeug:127.0.0.1 - - [06/Aug/2014 17:25:18] "GET /hello HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [06/Aug/2014 17:25:26] "GET /hello?name=Alex HTTP/1.1" 200 -
INFO:werkzeug: * Running on http://127.0.0.1:5000/
INFO:werkzeug:127.0.0.1 - - [06/Aug/2014 17:26:12] "GET /hello?name=Alex HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [06/Aug/2014 17:26:16] "GET /hello HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [06/Aug/2014 17:27:38] "GET /hello HTTP/1.1" 200 -
INFO:werkzeug: * Running on http://127.0.0.1:5000/
INFO:werkzeug:127.0.0.1 - - [06/Aug/2014 17:27:43] "GET /hello HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [06/Aug/2014 17:27:50] "GET /hello?name=Alex HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [06/Aug/2014 17:29:16] "GET /hello?name=Alex HTTP/1.1" 200 -
INFO:werkzeug: * Running on http://127.0.0.1:5000/
INFO:werkzeug:127.0.0.1 - - [06/Aug/2014 17:29:46] "GET /hello?name=Alex HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [06/Aug/2014 17:29:56] "GET /hello?name=Alex HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [06/Aug/2014 17:30:12] "GET /hello?name=Alex HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [06/Aug/2014 17:30:16] "GET /hello?name=Alex&greet=hola HTTP/1.1" 200 -
INFO:werkzeug: * Running on http://127.0.0.1:5000/
INFO:werkzeug:127.0.0.1 - - [06/Aug/2014 17:39:23] "GET /hello HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [06/Aug/2014 17:40:00] "POST /hello HTTP/1.1" 200 -
INFO:werkzeug: * Running on http://127.0.0.1:5000/
ERROR:__main__:Exception on / [GET]
Traceback (most recent call last):
  File "c:\Users\Alex\.virtualenvs\lpthw\lib\site-packages\flask\app.py", line 1817, in wsgi_app
    response = self.full_dispatch_request()
  File "c:\Users\Alex\.virtualenvs\lpthw\lib\site-packages\flask\app.py", line 1477, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "c:\Users\Alex\.virtualenvs\lpthw\lib\site-packages\flask\app.py", line 1381, in handle_user_exception
    reraise(exc_type, exc_value, tb)
  File "c:\Users\Alex\.virtualenvs\lpthw\lib\site-packages\flask\_compat.py", line 33, in reraise
    raise value
  File "c:\Users\Alex\.virtualenvs\lpthw\lib\site-packages\flask\app.py", line 1475, in full_dispatch_request
    rv = self.dispatch_request()
  File "c:\Users\Alex\.virtualenvs\lpthw\lib\site-packages\flask\app.py", line 1461, in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
  File "bin/app.py", line 22, in index
    return render_template('index.html', greeting=greeting)
  File "c:\Users\Alex\.virtualenvs\lpthw\lib\site-packages\flask\templating.py", line 127, in render_template
    return _render(ctx.app.jinja_env.get_or_select_template(template_name_or_list),
  File "c:\Users\Alex\.virtualenvs\lpthw\lib\site-packages\jinja2\environment.py", line 830, in get_or_select_template
    return self.get_template(template_name_or_list, parent, globals)
  File "c:\Users\Alex\.virtualenvs\lpthw\lib\site-packages\jinja2\environment.py", line 791, in get_template
    return self._load_template(name, self.make_globals(globals))
  File "c:\Users\Alex\.virtualenvs\lpthw\lib\site-packages\jinja2\environment.py", line 765, in _load_template
    template = self.loader.load(self, name, globals)
  File "c:\Users\Alex\.virtualenvs\lpthw\lib\site-packages\jinja2\loaders.py", line 113, in load
    source, filename, uptodate = self.get_source(environment, name)
  File "c:\Users\Alex\.virtualenvs\lpthw\lib\site-packages\flask\templating.py", line 64, in get_source
    raise TemplateNotFound(template)
jinja2.exceptions.TemplateNotFound: index.html
INFO:werkzeug:127.0.0.1 - - [06/Aug/2014 17:44:31] "GET / HTTP/1.1" 500 -
INFO:werkzeug: * Running on http://127.0.0.1:5000/
INFO:werkzeug:127.0.0.1 - - [06/Aug/2014 22:11:01] "GET / HTTP/1.1" 200 -
ERROR:__main__:Exception on /hello [GET]
Traceback (most recent call last):
  File "c:\Users\Alex\.virtualenvs\lpthw\lib\site-packages\flask\app.py", line 1817, in wsgi_app
    response = self.full_dispatch_request()
  File "c:\Users\Alex\.virtualenvs\lpthw\lib\site-packages\flask\app.py", line 1477, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "c:\Users\Alex\.virtualenvs\lpthw\lib\site-packages\flask\app.py", line 1381, in handle_user_exception
    reraise(exc_type, exc_value, tb)
  File "c:\Users\Alex\.virtualenvs\lpthw\lib\site-packages\flask\_compat.py", line 33, in reraise
    raise value
  File "c:\Users\Alex\.virtualenvs\lpthw\lib\site-packages\flask\app.py", line 1475, in full_dispatch_request
    rv = self.dispatch_request()
  File "c:\Users\Alex\.virtualenvs\lpthw\lib\site-packages\flask\app.py", line 1461, in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
  File "app.py", line 35, in hello
    return render_template('hello_form_block.html', name=name, greeting=greeting)
  File "c:\Users\Alex\.virtualenvs\lpthw\lib\site-packages\flask\templating.py", line 128, in render_template
    context, ctx.app)
  File "c:\Users\Alex\.virtualenvs\lpthw\lib\site-packages\flask\templating.py", line 110, in _render
    rv = template.render(context)
  File "c:\Users\Alex\.virtualenvs\lpthw\lib\site-packages\jinja2\environment.py", line 969, in render
    return self.environment.handle_exception(exc_info, True)
  File "c:\Users\Alex\.virtualenvs\lpthw\lib\site-packages\jinja2\environment.py", line 742, in handle_exception
    reraise(exc_type, exc_value, tb)
  File "c:\Users\Alex\.virtualenvs\lpthw\lib\site-packages\jinja2\_compat.py", line 36, in reraise
    raise value.with_traceback(tb)
  File "c:\Users\Alex\Dropbox\projects\learn_python_the_hard_way\projects\gothonweb\templates\layout.html", line 9, in <module>
    {% endblock %}
  File "c:\Users\Alex\.virtualenvs\lpthw\lib\site-packages\jinja2\environment.py", line 966, in render
    return concat(self.root_render_func(self.new_context(vars)))
  File "c:\Users\Alex\Dropbox\projects\learn_python_the_hard_way\projects\gothonweb\templates\hello_form_block.html", line 1, in <module>
    {% extends "layout.html" %}
  File "c:\Users\Alex\.virtualenvs\lpthw\lib\site-packages\jinja2\environment.py", line 742, in handle_exception
    reraise(exc_type, exc_value, tb)
  File "c:\Users\Alex\.virtualenvs\lpthw\lib\site-packages\jinja2\_compat.py", line 36, in reraise
    raise value.with_traceback(tb)
  File "c:\Users\Alex\Dropbox\projects\learn_python_the_hard_way\projects\gothonweb\templates\layout.html", line 9, in <module>
    {% endblock %}
  File "c:\Users\Alex\.virtualenvs\lpthw\lib\site-packages\jinja2\environment.py", line 459, in _parse
    return Parser(self, source, name, encode_filename(filename)).parse()
  File "c:\Users\Alex\.virtualenvs\lpthw\lib\site-packages\jinja2\parser.py", line 893, in parse
    result = nodes.Template(self.subparse(), lineno=1)
  File "c:\Users\Alex\.virtualenvs\lpthw\lib\site-packages\jinja2\parser.py", line 875, in subparse
    rv = self.parse_statement()
  File "c:\Users\Alex\.virtualenvs\lpthw\lib\site-packages\jinja2\parser.py", line 122, in parse_statement
    return getattr(self, 'parse_' + self.stream.current.value)()
  File "c:\Users\Alex\.virtualenvs\lpthw\lib\site-packages\jinja2\parser.py", line 229, in parse_block
    node.body = self.parse_statements(('name:endblock',), drop_needle=True)
  File "c:\Users\Alex\.virtualenvs\lpthw\lib\site-packages\jinja2\parser.py", line 162, in parse_statements
    self.fail_eof(end_tokens)
  File "c:\Users\Alex\.virtualenvs\lpthw\lib\site-packages\jinja2\parser.py", line 96, in fail_eof
    return self._fail_ut_eof(None, stack, lineno)
  File "c:\Users\Alex\.virtualenvs\lpthw\lib\site-packages\jinja2\parser.py", line 82, in _fail_ut_eof
    self.fail(' '.join(message), lineno)
  File "c:\Users\Alex\.virtualenvs\lpthw\lib\site-packages\jinja2\parser.py", line 51, in fail
    raise exc(msg, lineno, self.name, self.filename)
jinja2.exceptions.TemplateSyntaxError: Unexpected end of template. Jinja was looking for the following tags: 'endblock'. The innermost block that needs to be closed is 'block'.
INFO:werkzeug:127.0.0.1 - - [06/Aug/2014 22:11:06] "GET /hello HTTP/1.1" 500 -
INFO:werkzeug:127.0.0.1 - - [06/Aug/2014 22:12:06] "GET /hello HTTP/1.1" 200 -
ERROR:__main__:Exception on /hello [POST]
Traceback (most recent call last):
  File "c:\Users\Alex\.virtualenvs\lpthw\lib\site-packages\flask\app.py", line 1817, in wsgi_app
    response = self.full_dispatch_request()
  File "c:\Users\Alex\.virtualenvs\lpthw\lib\site-packages\flask\app.py", line 1477, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "c:\Users\Alex\.virtualenvs\lpthw\lib\site-packages\flask\app.py", line 1381, in handle_user_exception
    reraise(exc_type, exc_value, tb)
  File "c:\Users\Alex\.virtualenvs\lpthw\lib\site-packages\flask\_compat.py", line 33, in reraise
    raise value
  File "c:\Users\Alex\.virtualenvs\lpthw\lib\site-packages\flask\app.py", line 1475, in full_dispatch_request
    rv = self.dispatch_request()
  File "c:\Users\Alex\.virtualenvs\lpthw\lib\site-packages\flask\app.py", line 1461, in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
  File "app.py", line 31, in hello
    return render_template('hello.html_block', name=name, greeting=greeting)
  File "c:\Users\Alex\.virtualenvs\lpthw\lib\site-packages\flask\templating.py", line 127, in render_template
    return _render(ctx.app.jinja_env.get_or_select_template(template_name_or_list),
  File "c:\Users\Alex\.virtualenvs\lpthw\lib\site-packages\jinja2\environment.py", line 830, in get_or_select_template
    return self.get_template(template_name_or_list, parent, globals)
  File "c:\Users\Alex\.virtualenvs\lpthw\lib\site-packages\jinja2\environment.py", line 791, in get_template
    return self._load_template(name, self.make_globals(globals))
  File "c:\Users\Alex\.virtualenvs\lpthw\lib\site-packages\jinja2\environment.py", line 765, in _load_template
    template = self.loader.load(self, name, globals)
  File "c:\Users\Alex\.virtualenvs\lpthw\lib\site-packages\jinja2\loaders.py", line 113, in load
    source, filename, uptodate = self.get_source(environment, name)
  File "c:\Users\Alex\.virtualenvs\lpthw\lib\site-packages\flask\templating.py", line 64, in get_source
    raise TemplateNotFound(template)
jinja2.exceptions.TemplateNotFound: hello.html_block
INFO:werkzeug:127.0.0.1 - - [06/Aug/2014 22:12:16] "POST /hello HTTP/1.1" 500 -
ERROR:__main__:Exception on /hello [POST]
Traceback (most recent call last):
  File "c:\Users\Alex\.virtualenvs\lpthw\lib\site-packages\flask\app.py", line 1817, in wsgi_app
    response = self.full_dispatch_request()
  File "c:\Users\Alex\.virtualenvs\lpthw\lib\site-packages\flask\app.py", line 1477, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "c:\Users\Alex\.virtualenvs\lpthw\lib\site-packages\flask\app.py", line 1381, in handle_user_exception
    reraise(exc_type, exc_value, tb)
  File "c:\Users\Alex\.virtualenvs\lpthw\lib\site-packages\flask\_compat.py", line 33, in reraise
    raise value
  File "c:\Users\Alex\.virtualenvs\lpthw\lib\site-packages\flask\app.py", line 1475, in full_dispatch_request
    rv = self.dispatch_request()
  File "c:\Users\Alex\.virtualenvs\lpthw\lib\site-packages\flask\app.py", line 1461, in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
  File "app.py", line 31, in hello
    return render_template('hello_block.html', name=name, greeting=greeting)
  File "c:\Users\Alex\.virtualenvs\lpthw\lib\site-packages\flask\templating.py", line 127, in render_template
    return _render(ctx.app.jinja_env.get_or_select_template(template_name_or_list),
  File "c:\Users\Alex\.virtualenvs\lpthw\lib\site-packages\jinja2\environment.py", line 830, in get_or_select_template
    return self.get_template(template_name_or_list, parent, globals)
  File "c:\Users\Alex\.virtualenvs\lpthw\lib\site-packages\jinja2\environment.py", line 791, in get_template
    return self._load_template(name, self.make_globals(globals))
  File "c:\Users\Alex\.virtualenvs\lpthw\lib\site-packages\jinja2\environment.py", line 765, in _load_template
    template = self.loader.load(self, name, globals)
  File "c:\Users\Alex\.virtualenvs\lpthw\lib\site-packages\jinja2\loaders.py", line 113, in load
    source, filename, uptodate = self.get_source(environment, name)
  File "c:\Users\Alex\.virtualenvs\lpthw\lib\site-packages\flask\templating.py", line 64, in get_source
    raise TemplateNotFound(template)
jinja2.exceptions.TemplateNotFound: hello.html_block
INFO:werkzeug:127.0.0.1 - - [06/Aug/2014 22:12:39] "POST /hello HTTP/1.1" 500 -
INFO:werkzeug:127.0.0.1 - - [06/Aug/2014 22:12:41] "GET /hello HTTP/1.1" 200 -
ERROR:__main__:Exception on /hello [POST]
Traceback (most recent call last):
  File "c:\Users\Alex\.virtualenvs\lpthw\lib\site-packages\flask\app.py", line 1817, in wsgi_app
    response = self.full_dispatch_request()
  File "c:\Users\Alex\.virtualenvs\lpthw\lib\site-packages\flask\app.py", line 1477, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "c:\Users\Alex\.virtualenvs\lpthw\lib\site-packages\flask\app.py", line 1381, in handle_user_exception
    reraise(exc_type, exc_value, tb)
  File "c:\Users\Alex\.virtualenvs\lpthw\lib\site-packages\flask\_compat.py", line 33, in reraise
    raise value
  File "c:\Users\Alex\.virtualenvs\lpthw\lib\site-packages\flask\app.py", line 1475, in full_dispatch_request
    rv = self.dispatch_request()
  File "c:\Users\Alex\.virtualenvs\lpthw\lib\site-packages\flask\app.py", line 1461, in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
  File "app.py", line 31, in hello
    return render_template('hello_block.html', name=name, greeting=greeting)
  File "c:\Users\Alex\.virtualenvs\lpthw\lib\site-packages\flask\templating.py", line 127, in render_template
    return _render(ctx.app.jinja_env.get_or_select_template(template_name_or_list),
  File "c:\Users\Alex\.virtualenvs\lpthw\lib\site-packages\jinja2\environment.py", line 830, in get_or_select_template
    return self.get_template(template_name_or_list, parent, globals)
  File "c:\Users\Alex\.virtualenvs\lpthw\lib\site-packages\jinja2\environment.py", line 791, in get_template
    return self._load_template(name, self.make_globals(globals))
  File "c:\Users\Alex\.virtualenvs\lpthw\lib\site-packages\jinja2\environment.py", line 765, in _load_template
    template = self.loader.load(self, name, globals)
  File "c:\Users\Alex\.virtualenvs\lpthw\lib\site-packages\jinja2\loaders.py", line 113, in load
    source, filename, uptodate = self.get_source(environment, name)
  File "c:\Users\Alex\.virtualenvs\lpthw\lib\site-packages\flask\templating.py", line 64, in get_source
    raise TemplateNotFound(template)
jinja2.exceptions.TemplateNotFound: hello.html_block
INFO:werkzeug:127.0.0.1 - - [06/Aug/2014 22:12:48] "POST /hello HTTP/1.1" 500 -
INFO:werkzeug: * Running on http://127.0.0.1:5000/
INFO:werkzeug:127.0.0.1 - - [06/Aug/2014 22:13:02] "GET /hello HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [06/Aug/2014 22:13:06] "POST /hello HTTP/1.1" 200 -
INFO:werkzeug: * Running on http://127.0.0.1:5000/
INFO:werkzeug: * Restarting with reloader
INFO:werkzeug:127.0.0.1 - - [06/Aug/2014 22:14:45] "GET /hello HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [06/Aug/2014 22:14:49] "POST /hello HTTP/1.1" 200 -
INFO:werkzeug: * Detected change in 'app.py', reloading
INFO:werkzeug: * Restarting with reloader
INFO:werkzeug:127.0.0.1 - - [06/Aug/2014 22:14:59] "GET /hello HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [06/Aug/2014 22:15:38] "GET /hello HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [06/Aug/2014 22:15:52] "GET /hello HTTP/1.1" 200 -
INFO:werkzeug: * Detected change in 'app.py', reloading
INFO:werkzeug: * Restarting with reloader
INFO:werkzeug:127.0.0.1 - - [06/Aug/2014 22:16:44] "GET /hello HTTP/1.1" 500 -
INFO:werkzeug:127.0.0.1 - - [06/Aug/2014 22:16:44] "GET /hello?__debugger__=yes&cmd=resource&f=style.css HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [06/Aug/2014 22:16:44] "GET /hello?__debugger__=yes&cmd=resource&f=jquery.js HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [06/Aug/2014 22:16:44] "GET /hello?__debugger__=yes&cmd=resource&f=debugger.js HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [06/Aug/2014 22:16:44] "GET /hello?__debugger__=yes&cmd=resource&f=ubuntu.ttf HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [06/Aug/2014 22:16:44] "GET /hello?__debugger__=yes&cmd=resource&f=console.png HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [06/Aug/2014 22:16:44] "GET /hello?__debugger__=yes&cmd=resource&f=source.png HTTP/1.1" 200 -
INFO:werkzeug:127.0.0.1 - - [06/Aug/2014 22:16:44] "GET /hello?__debugger__=yes&cmd=resource&f=console.png HTTP/1.1" 200 -
INFO:werkzeug: * Detected change in 'app.py', reloading
INFO:werkzeug: * Restarting with reloader
INFO:werkzeug:127.0.0.1 - - [06/Aug/2014 22:16:50] "GET /hello HTTP/1.1" 200 -
INFO:werkzeug: * Detected change in 'app.py', reloading
INFO:werkzeug: * Restarting with reloader
